inspectit-gepard-agent/src/main/java/rocks/inspectit/gepard/agent/instrumentation/hook/MethodHookState.java

```diff
@@ -73,16 +73,13 @@ public int updateHooks(Class<?> clazz, ClassHookConfiguration classConfiguration
     classConfiguration
         .asMap()
         .forEach(
-            (method, active) -> {
-              // Currently always true, later we should compare the current with the new config
-              if (active) {
-                String signature = getSignature(method);
-                Optional<MethodHook> maybeHook = getCurrentHook(clazz, signature);
-                if (maybeHook.isEmpty()) {
-                  MethodHook hook = MethodHookGenerator.createHook(method);
-                  setHook(clazz, signature, hook);
-                  operationCounter.addAndGet(1);
-                }
+            (method, newConfig) -> {
+              String signature = getSignature(method);
+              MethodHookConfiguration currentConfig = getCurrentHookConfiguration(clazz, signature);
+              if (!newConfig.equals(currentConfig)) {
+                MethodHook hook = MethodHookGenerator.createHook(newConfig);
```

Comment:
I could imagine some side effects, if we change the configuration between the onEnter() and onExit() calls of one method hook. I believe it would be more safe to overwrite the whole method hook. But good thought!